storage/remote/queue_manager.go

```diff
@@ -737,26 +812,70 @@ func (s *shards) stop() {
 	}
 }
 
-// enqueue a sample.  If we are currently in the process of shutting down or resharding,
+// bufferPendingSamples listens for incoming pending samples that are caused
+// due to closing of queues. They are stored in a temporary buffer before
+// being appended to the new shards.
+func (s *shards) bufferPendingSamples() {
+	var (
+		wg             sync.WaitGroup
+		bufLock        sync.Mutex
+		buffer         []sample
+		oldShardsCount = len(*s.queueBuf)
+	)
+	if oldShardsCount == 0 {
+		return
+	}
+	wg.Add(oldShardsCount)
+	for i := 0; i < oldShardsCount; i++ {
+		go func(i int) {
+			buf := *s.queueBuf
+			temp, ok := <-buf[i]
+			if !ok {
+				wg.Done()
+				return
+			}
+			bufLock.Lock()
+			buffer = append(buffer, temp...)
+			bufLock.Unlock()
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+	sort.SliceStable(buffer, func(i, j int) bool {
+		return buffer[i].t < buffer[j].t
+	})
+	s.qm.RedirectPrevSamples(buffer)
+}
+
+// updateQueues replaces the old queues with the new ones. Post replacement,
```

Comment:
the comment here seems incorrect?

Reply:
this almost seems better to just inline in the reshard block, can't think of a name that's actually descriptive

Reply:
Yeah, sorry, missed changing the naming of the function.